From 015f1f45695f32d6f955da124fa29036b6150359 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Miroslav=20Mr=C3=A1z?= <mraz.miro@seznam.cz>
Date: Sat, 25 Jul 2026 22:53:18 +0200
Subject: [PATCH 1/9] media: dt-bindings: media: add qcom,sm7150-venus
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add the binding for the Venus video decode/encode accelerator found on the
Qualcomm SM7150 SoC. The IP is an IRIS1-generation core with the classic
V4/AR50 register layout. In addition to the video core it carries a CVP
co-processor whose power domain and clocks are exposed as the "cvp" domain
and the "cvp_core"/"cvp_bus" clocks; the firmware requires them at boot.

Signed-off-by: Miroslav Mráz <mraz.miro@seznam.cz>
---
 .../bindings/media/qcom,sm7150-venus.yaml     | 134 ++++++++++++++++++
 1 file changed, 134 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/media/qcom,sm7150-venus.yaml

diff --git a/Documentation/devicetree/bindings/media/qcom,sm7150-venus.yaml b/Documentation/devicetree/bindings/media/qcom,sm7150-venus.yaml
new file mode 100644
index 000000000000..6786dc115b52
--- /dev/null
+++ b/Documentation/devicetree/bindings/media/qcom,sm7150-venus.yaml
@@ -0,0 +1,134 @@
+# SPDX-License-Identifier: (GPL-2.0 OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/media/qcom,sm7150-venus.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Qualcomm SM7150 Venus video encode and decode accelerator
+
+maintainers:
+  - Miroslav Mráz <mraz.miro@seznam.cz>
+
+description: |
+  The Venus IP is a video encode and decode accelerator present
+  on Qualcomm platforms
+
+allOf:
+  - $ref: qcom,venus-common.yaml#
+
+properties:
+  compatible:
+    const: qcom,sm7150-venus
+
+  power-domains:
+    maxItems: 4
+
+  power-domain-names:
+    items:
+      - const: venus
+      - const: vcodec0
+      - const: cvp
+      - const: cx
+
+  clocks:
+    maxItems: 7
+
+  clock-names:
+    items:
+      - const: core
+      - const: iface
+      - const: bus
+      - const: vcodec0_core
+      - const: vcodec0_bus
+      - const: cvp_core
+      - const: cvp_bus
+
+  iommus:
+    maxItems: 1
+
+  memory-region:
+    maxItems: 1
+
+  interconnects:
+    maxItems: 2
+
+  interconnect-names:
+    items:
+      - const: cpu-cfg
+      - const: video-mem
+
+  operating-points-v2: true
+  opp-table:
+    type: object
+
+  video-decoder:
+    type: object
+
+    properties:
+      compatible:
+        const: venus-decoder
+
+    required:
+      - compatible
+
+    deprecated: true
+    additionalProperties: false
+
+  video-encoder:
+    type: object
+
+    properties:
+      compatible:
+        const: venus-encoder
+
+    required:
+      - compatible
+
+    deprecated: true
+    additionalProperties: false
+
+required:
+  - compatible
+  - power-domain-names
+  - iommus
+
+unevaluatedProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/interrupt-controller/arm-gic.h>
+    #include <dt-bindings/clock/qcom,sm7150-videocc.h>
+    #include <dt-bindings/power/qcom,rpmhpd.h>
+
+    venus: video-codec@aa00000 {
+        compatible = "qcom,sm7150-venus";
+        reg = <0x0aa00000 0xff000>;
+        interrupts = <GIC_SPI 174 IRQ_TYPE_LEVEL_HIGH>;
+        power-domains = <&videocc VENUS_GDSC>,
+                        <&videocc VCODEC0_GDSC>,
+                        <&videocc VCODEC1_GDSC>,
+                        <&rpmhpd RPMHPD_CX>;
+        power-domain-names = "venus", "vcodec0", "cvp", "cx";
+        clocks = <&videocc VIDEOCC_MVSC_CORE_CLK>,
+                 <&videocc VIDEOCC_VENUS_AHB_CLK>,
+                 <&videocc VIDEOCC_MVSC_CTL_AXI_CLK>,
+                 <&videocc VIDEOCC_MVS0_CORE_CLK>,
+                 <&videocc VIDEOCC_MVS0_AXI_CLK>,
+                 <&videocc VIDEOCC_MVS1_CORE_CLK>,
+                 <&videocc VIDEOCC_MVS1_AXI_CLK>;
+        clock-names = "core", "iface", "bus",
+                      "vcodec0_core", "vcodec0_bus",
+                      "cvp_core", "cvp_bus";
+        iommus = <&apps_smmu 0x1080 0x60>;
+        memory-region = <&venus_mem>;
+        operating-points-v2 = <&venus_opp_table>;
+
+        venus_opp_table: opp-table {
+            compatible = "operating-points-v2";
+
+            opp-240000000 {
+                opp-hz = /bits/ 64 <240000000>;
+                required-opps = <&rpmhpd_opp_low_svs>;
+            };
+        };
+    };
-- 
2.55.0

